### PR TITLE
Add no_option test for save-image-edit

### PR DESCRIPTION
**Pull request by kylazhang**
1. Add no_option test and add xml back/recover step
2. Also update cfg format for save-image-define

Signed-off-by: Kylazhang <weizhan@redhat.com>

### CODE REVIEW

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_save_image_define.cfg [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
     kill_vm_on_error = "no"
     variants:
-       - running:
-           restore_state = "running"
-       - paused:
-           restore_state = "paused"
+        - running:
+            restore_state = "running"
+        - paused:
+            restore_state = "paused"
```
Comment (balamuruhans) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_save_image_define.cfg]: we can have this fix as a separate commit.

**Comment by iccaszhulili** [comment]
rh-caselink retest please

File: libvirt/tests/src/virsh_cmd/domain/virsh_save_image_edit.py [hunk 2]
```diff
@@ -97,6 +102,8 @@ def vm_state_check():
 
     finally:
         # cleanup
+        vm_backup.sync()
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_save_image_edit.py]: Better destroy VM before do vm backup

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_save_image_define.cfg (hunk 1 above)
Comment (kylazhang) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_save_image_define.cfg]: As I need add another option, so change them at same time.

File: libvirt/tests/src/virsh_cmd/domain/virsh_save_image_edit.py (hunk 2 above)
Comment (kylazhang) [reply libvirt/tests/src/virsh_cmd/domain/virsh_save_image_edit.py]: no needed, sync will update the inactive xml.

**Pull request merged**